timestamp-impl/src/main/java/com/palantir/timestamp/PersistentTimestampService.java

```diff
@@ -60,7 +62,7 @@ public long getFreshTimestamp() {
     }
 
     @Override
-    public synchronized TimestampRange getFreshTimestamps(int numTimestampsRequested) {
+    public TimestampRange getFreshTimestamps(int numTimestampsRequested) {
         int numTimestampsToHandOut = cleanUpTimestampRequest(numTimestampsRequested);
         TimestampRange handedOut = availableTimestamps.handOut(numTimestampsToHandOut);
         asynchronouslyRefreshBuffer();
```

Comment:
note to reviewers: this was the major source of contention via the `synchronized` on the method. There is now no need for method level locking as `cleanUpTimestampRequest` is a static that checks argument, `availableTimestamps.handOut` does its own locking, and `asynchronouslyRefreshBuffer()` now handles ensuring only a single refresh task is in-flight at a time.